File: backend/internal/constants/order.go
package constants

// OrderType 单据类型枚举。
const (
	OrderTypeInbound        = "inbound"
	OrderTypeOutbound       = "outbound"
	OrderTypeTransfer       = "transfer"
	OrderTypeInventoryCheck = "inventory_check"
)

// OrderTypeValues 全部单据类型值。
var OrderTypeValues = []string{OrderTypeInbound, OrderTypeOutbound, OrderTypeTransfer, OrderTypeInventoryCheck}

// OrderStatus 单据状态枚举。
const (
	OrderStatusDraft      = "draft"
	OrderStatusSubmitted  = "submitted"
	OrderStatusProcessing = "processing"
	OrderStatusCompleted  = "completed"
	OrderStatusCancelled  = "cancelled"
)

// OrderStatusValues 全部单据状态值。
var OrderStatusValues = []string{OrderStatusDraft, OrderStatusSubmitted, OrderStatusProcessing, OrderStatusCompleted, OrderStatusCancelled}

// StockOpType 库存操作类型枚举。
const (
	StockOpInbound  = "inbound"
	StockOpOutbound = "outbound"
	StockOpAdjust   = "adjust"
	StockOpTransfer = "transfer"
)

// IsValidOrderType 校验单据类型。
func IsValidOrderType(s string) bool {
	for _, v := range OrderTypeValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidOrderStatus 校验单据状态。
func IsValidOrderStatus(s string) bool {
	for _, v := range OrderStatusValues {
		if v == s {
			return true
		}
	}
	return false
}

File: frontend/src/constants/order.ts
// 单据类型/状态枚举（与后端 backend/internal/constants/order.go 保持一致）
export const OrderType = {
  INBOUND: 'inbound',
  OUTBOUND: 'outbound',
  TRANSFER: 'transfer',
  INVENTORY_CHECK: 'inventory_check',
} as const

export const OrderTypeText: Record<string, string> = {
  [OrderType.INBOUND]: '入库',
  [OrderType.OUTBOUND]: '出库',
  [OrderType.TRANSFER]: '调拨',
  [OrderType.INVENTORY_CHECK]: '盘点',
}

export const OrderStatus = {
  DRAFT: 'draft',
  SUBMITTED: 'submitted',
  PROCESSING: 'processing',
  COMPLETED: 'completed',
  CANCELLED: 'cancelled',
} as const

export const OrderStatusText: Record<string, string> = {
  [OrderStatus.DRAFT]: '草稿',
  [OrderStatus.SUBMITTED]: '待审批',
  [OrderStatus.PROCESSING]: '处理中',
  [OrderStatus.COMPLETED]: '已完成',
  [OrderStatus.CANCELLED]: '已取消',
}

export const OrderTypeOptions = Object.entries(OrderTypeText).map(([value, label]) => ({ label, value }))
export const OrderStatusOptions = Object.entries(OrderStatusText).map(([value, label]) => ({ label, value }))
